balancer_wrapper.go

```diff
@@ -261,7 +261,9 @@ func (acbw *acBalancerWrapper) updateState(s connectivity.State, err error) {
 }
 
 func (acbw *acBalancerWrapper) String() string {
-	return fmt.Sprintf("SubConn(id:%d)", acbw.ac.channelz.ID)
+	acbw.ac.mu.Lock()
+	defer acbw.ac.mu.Unlock()
+	return fmt.Sprintf("SubConn(id:%d;addr:%s)", acbw.ac.channelz.ID, acbw.ac.curAddr.Addr)
```

Comment:
The impossibility of finding out which `Addr` a given `SubConn` my balancer is using is something I stumbled upon today, so not sure if this PR is the right place for it, but since it's also related to a Stringer, I figured maybe?
 
But this requires a mutex lock I believe, because  `UpdateAddresses` can `ac.resetTransport` and that can change `curAddr.`... So not sure if it's worth it, especially in a `String`? 

I'd love to know how one can find out the exact `Addr` a given `balancer.SubConn` is using if you know how to do it, since I see  @townba removed `curAddr` from the `channelz.Infof` in `ac.updateAddrs` earlier this year.

Reply:
I have now discovered I could run a localhost Server and query the channelz directly using `grpc_channelz_v1`, which means I can query the target of a subchannel using this, I've reverted my commit since I don't feel good about adding a mutex lock in a `Stringer`.